src/main/java/christmas/domain/order/OrderValidator.java

```diff
@@ -0,0 +1,77 @@
+package christmas.domain.order;
+
+import christmas.domain.Menu;
+import christmas.domain.MenuType;
+import christmas.domain.exception.DuplicateMenuException;
+import christmas.domain.exception.ExceedTotalMenuCountException;
+import christmas.domain.exception.OnlyBeverageMenuException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class OrderValidator {
+
+    private static final int TOTAL_ORDER_COUNT_LIMIT = 20;
+
+    /**
+     * @throws IllegalArgumentException:
+     * @throws DuplicateMenuException:        중복 메뉴가 존재하는 경우
+     * @throws OnlyBeverageMenuException:     음료수만 주문한 경우
+     * @throws ExceedTotalMenuCountException: 주문 개수가 20개 초과인 경우
+     */
+    public static void validate(List<OrderItem> orderMenus) {
+        validateDuplicateMenu(orderMenus);
+        validateOnlyDrinkMenu(orderMenus);
+        validateTotalOrderCount(orderMenus);
+    }
+
+    private static void validateDuplicateMenu(List<OrderItem> orderMenus) {
+        Set<Menu> uniqueOrderMenus = createUniqueOrderMenus(orderMenus);
+
+        if (orderMenus.size() != uniqueOrderMenus.size()) {
+            throw new DuplicateMenuException();
+        }
+    }
+
+    private static Set<Menu> createUniqueOrderMenus(List<OrderItem> orderMenus) {
+        return orderMenus.stream()
+                .map(OrderItem::getMenuName)
+                .map(Menu::find)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toSet());
+    }
+
+    private static void validateOnlyDrinkMenu(List<OrderItem> orderMenus) {
+        boolean allMenuBeverage = isAllMenuBeverage(orderMenus);
+
+        if (allMenuBeverage) {
+            throw new OnlyBeverageMenuException();
+        }
+    }
+
+    private static boolean isAllMenuBeverage(List<OrderItem> orderMenus) {
+        return orderMenus.stream()
+                .map(OrderItem::getMenuName)
+                .map(Menu::find)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(Menu::getMenuType)
+                .allMatch(menuType -> menuType == MenuType.BEVERAGE);
```

Comment:
`OrderItem`은 이미 @june-777 님이 검증 로직을 잘 짜 두셔서 항상 검증된 값들만 가지는 객체로 존재할 수 있게 되는데요.
때문에 이름을 가져와 존재하지 않는 메뉴들을 걸러내는 일련의 과정들은 불필요한 작업이라는 생각이 들어요.
또한, 이미 @june-777 님이 `OrderItem`에서도 메뉴 타입을 체크하는 메서드를 만들어 두었기 때문에 `OrderItem`만 이용해서 스트림을 작성할 수 있을 것 같습니다.
```suggestion
        return orderMenus.stream()
                .allMatch(orderItem -> orderItem.isSameMenuType(MenuType.BEVERAGE));
```

Reply:
예리한 지적이십니다.. 맞는 말씀이에요 불필요한 과정을 한 번 더 하고 있었던거네요 😂 좋은 말씀 감사합니다 :)